src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c

```diff
@@ -31,53 +32,367 @@ const ActorInit Obj_Mure_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_808D87BC[] = {
+static f32 sZClip[] = {
+    1600.0f, 1600.0f, 1000.0f, 1000.0f, 1000.0f,
+};
+
+static s32 sMaxChildSpawns[] = {
+    12,
+    9,
+    8,
+    0,
+};
+
+static s16 sSpawnActorIds[] = {
+    ACTOR_EN_KUSA, 0, ACTOR_EN_FISH, ACTOR_EN_INSECT, ACTOR_EN_BUTTE,
+};
+
+static s16 sSpawnParams[] = {
+    0, 2, -1, 0, -1,
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 1200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 1200, ICHAIN_STOP),
 };
 
-#endif
+typedef enum {
+    /* 0 */ OBJMURE_TYPE_GRASS,
+    /* 1 */ OBJMURE_TYPE_UNDEFINED,
+    /* 2 */ OBJMURE_TYPE_FISH,
+    /* 3 */ OBJMURE_TYPE_BUGS,
+    /* 4 */ OBJMURE_TYPE_BUTTERFLY
+} ObjMureType;
+
+typedef enum {
+    /* 0 */ OBJMURE_CHILD_STATE_0,
+    /* 1 */ OBJMURE_CHILD_STATE_DEAD,
+    /* 2 */ OBJMURE_CHILD_STATE_2
+} ObjMureChildState;
+
+s32 func_808D78D0(ObjMure* this, GlobalContext* globalCtx) {
+    if (this->type == OBJMURE_TYPE_FISH || this->type == OBJMURE_TYPE_BUGS || this->type == OBJMURE_TYPE_BUTTERFLY) {
+        Actor_ProcessInitChain(&this->actor, sInitChain);
+    } else {
+        return false;
+    }
+    return true;
+}
+
+s32 func_808D7928(ObjMure* this, GlobalContext* globalCtx) {
+    if (!func_808D78D0(this, globalCtx)) {
+        return false;
+    }
+    return true;
+}
 
-extern InitChainEntry D_808D87BC[];
+void ObjMure_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjMure* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure/func_808D78D0.s")
+    this->chNum = OBJ_MURE_GET_CHNUM(this->actor.params);
+    this->ptn = OBJ_MURE_GET_PTN(this->actor.params);
+    this->svNum = OBJ_MURE_GET_SVNUM(this->actor.params);
+    this->type = OBJ_MURE_GET_TYPE(this->actor.params);
+    if (this->ptn >= 4) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    if (this->type >= 5) {
```

Comment:
You could add a `OBJMURE_TYPE_MAX` to the type enum above and then do:
```suggestion
    if (this->type >= OBJMURE_TYPE_MAX) {
```

Reply:
I agree with this. I'll approve and merge once this is done.